src/main/java/de/tum/in/www1/artemis/web/rest/lecture/OnlineUnitResource.java

```diff
@@ -0,0 +1,203 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import de.tum.in.www1.artemis.domain.Lecture;
+import de.tum.in.www1.artemis.domain.lecture.OnlineUnit;
+import de.tum.in.www1.artemis.repository.LectureRepository;
+import de.tum.in.www1.artemis.repository.OnlineUnitRepository;
+import de.tum.in.www1.artemis.security.Role;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.web.rest.dto.OnlineResourceDTO;
+import de.tum.in.www1.artemis.web.rest.errors.ConflictException;
+import de.tum.in.www1.artemis.web.rest.errors.InternalServerErrorException;
+import de.tum.in.www1.artemis.web.rest.util.HeaderUtil;
+
+@RestController
+@RequestMapping("/api")
+public class OnlineUnitResource {
+
+    @Value("${jhipster.clientApp.name}")
+    private String applicationName;
+
+    private final Logger log = LoggerFactory.getLogger(OnlineUnitResource.class);
+
+    private static final String ENTITY_NAME = "onlineUnit";
+
+    private final OnlineUnitRepository onlineUnitRepository;
+
+    private final LectureRepository lectureRepository;
+
+    private final AuthorizationCheckService authorizationCheckService;
+
+    public OnlineUnitResource(LectureRepository lectureRepository, AuthorizationCheckService authorizationCheckService, OnlineUnitRepository onlineUnitRepository) {
+        this.lectureRepository = lectureRepository;
+        this.authorizationCheckService = authorizationCheckService;
+        this.onlineUnitRepository = onlineUnitRepository;
+    }
+
+    /**
+     * GET lectures/:lectureId/online-units/:onlineUnitId: gets the online unit with the specified id
+     *
+     * @param onlineUnitId the id of the onlineUnit to retrieve
+     * @param lectureId the id of the lecture to which the unit belongs
+     * @return the ResponseEntity with status 200 (OK) and with body the online unit, or with status 404 (Not Found)
+     */
+    @GetMapping("lectures/{lectureId}/online-units/{onlineUnitId}")
+    @PreAuthorize("hasRole('EDITOR')")
+    public ResponseEntity<OnlineUnit> getOnlineUnit(@PathVariable Long onlineUnitId, @PathVariable Long lectureId) {
+        log.debug("REST request to get onlineUnit : {}", onlineUnitId);
+        var onlineUnit = onlineUnitRepository.findByIdElseThrow(onlineUnitId);
+        if (onlineUnit.getLecture() == null || onlineUnit.getLecture().getCourse() == null) {
+            throw new ConflictException("Lecture unit must be associated to a lecture of a course", "onlineUnit", "lectureOrCourseMissing");
+        }
+        if (!onlineUnit.getLecture().getId().equals(lectureId)) {
+            throw new ConflictException("Requested lecture unit is not part of the specified lecture", "onlineUnit", "lectureIdMismatch");
+        }
```

Comment:
I guess the conflict exception does not fit here:
`Conflicts are most likely to occur in response to a [PUT](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT) request. For example, you may get a 409 response when uploading a file that is older than the existing one on the server, resulting in a version control conflict.`

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/409

Reply:
You are technically correct. If we change this, when in the whole codebase and in a separate PR. Would you mind bringing it up in the next developer meeting?